client/tokenized-express-checkout/index.js

```diff
@@ -417,6 +380,14 @@ jQuery( ( $ ) => {
 		init: async () => {
 			// on product pages, we should be able to have `getExpressCheckoutData( 'product' )` from the backend,
 			// which saves us some AJAX calls.
+			if (
+				getExpressCheckoutData( 'button_context' ) === 'product' &&
+				getExpressCheckoutData( 'product' )?.product_type === 'bundle'
+			) {
+				// server-side data for bundled products is not reliable.
+				wcpayExpressCheckoutParams.product = undefined;
```

Comment:
This should be self-explanatory - it looks like the data truly cannot be relied upon (`needs_shipping` & totals are off)